### PR TITLE
Implement additional caches

### PR DESCRIPTION
**Pull request by forki**
as described in #1329

### CODE REVIEW

File: src/Paket.Core/Cache.fs [hunk 1]
```diff
@@ -2,11 +2,47 @@
...
+    | CurrentVersion
 
 type Cache = 
     { Location : string
-      CacheType : CacheType }
\ No newline at end of file
+      CacheType : CacheType option }
+
+    static member Parse(line : string) =
+        let sourceRegex = Regex("cache[ ]*[\"]([^\"]*)[\"]", RegexOptions.IgnoreCase)
```
Comment (mariusschulz) [inline src/Paket.Core/Cache.fs]: There's no need for the first and last character class here: `"cache[ ]*\"([^\"]*)\""` should be fine.

**Comment by adamchester** [comment]
Just FYI I gave this a few quick tests and it seems to behave as I expected. I wondered the first time if the packages missing from the local cache deserved a "something went wrong" message.

**Comment by forki** [comment]
Thx. Will continue this later today. There is a lot missing

File: src/Paket.Core/Cache.fs (hunk 1 above)
Comment (forki) [inline src/Paket.Core/Cache.fs]: it's copied from nuget packages sources and I might add passwords here

**Comment by forki** [comment]
> I wondered the first time if the packages missing from the local cache deserved a "something went wrong" message.

what does this mean?

**Comment by adamchester** [comment]
I'm not in front of a computer right now, so I can't get the exact message, but it gave a message that indicated something was wrong when the package wasn't in the cache. It seemed a little excessive to say something was "wrong".

**Pull request merged**